src/EnergyPlus/HeatBalanceSurfaceManager.cc

```diff
@@ -8874,4 +8835,59 @@ void GatherComponentLoadsSurfAbsFact(EnergyPlusData &state)
     }
 }
 
+void InitSurfacePropertyViewFactors(EnergyPlusData &state)
+{
+
+    // purpuse:
+    //   Initializes sky and ground surfaces view factors of exterior surfaces
+    //   used by SurfaceProperty:LocalEnvironment
+    //   view factors are constant hence should be set only once
+
+    if (!state.dataGlobal->AnyLocalEnvironmentsInModel) {
+        return;
+    }
+    if (!state.dataHeatBalSurfMgr->InitSurfaceHeatBalancefirstTime) {
+        return;
+    }
+    for (int SurfNum = 1; SurfNum <= state.dataSurface->TotSurfaces; ++SurfNum) {
+        if (!state.dataSurface->SurfHasSurroundingSurfProperties(SurfNum)) continue;
+        auto &Surface = state.dataSurface->Surface(SurfNum);
+        int SrdSurfsNum = state.dataSurface->SurfSurroundingSurfacesNum(SurfNum);
+        auto &SrdSurfsProperty = state.dataSurface->SurroundingSurfsProperty(SrdSurfsNum);
+        Real64 SrdSurfsViewFactor = 0;
+
+        SrdSurfsViewFactor += SrdSurfsProperty.SkyViewFactor;
+        SrdSurfsViewFactor += SrdSurfsProperty.GroundViewFactor;
+        for (int SrdSurfNum = 1; SrdSurfNum <= SrdSurfsProperty.TotSurroundingSurface; SrdSurfNum++) {
+            SrdSurfsViewFactor += SrdSurfsProperty.SurroundingSurfs(SrdSurfNum).ViewFactor;
+        }
+        // Check if the sum of all defined view factors > 1.0
+        if (SrdSurfsViewFactor > 1.0) {
+            ShowSevereError(state, "Illegal surrounding surfaces view factors for " + Surface.Name + ".");
+            ShowContinueError(state, " The sum of sky, ground, and all surrounding surfaces view factors should be less than or equal to 1.0.");
+        }
+        if (SrdSurfsProperty.SkyViewFactor > 0.0 && SrdSurfsProperty.GroundViewFactor > 0.0) {
+            // If both surface sky and ground view factor defined, overwrite with the defined value
+            Surface.ViewFactorSkyIR = SrdSurfsProperty.SkyViewFactor;
+            Surface.ViewFactorGroundIR = SrdSurfsProperty.GroundViewFactor;
+        } else if (SrdSurfsProperty.SkyViewFactor > 0.0 && SrdSurfsProperty.GroundViewFactor == 0.0) {
+            // If only sky view factor defined, ground view factor = 1 - all other defined view factors.
+            Surface.ViewFactorSkyIR = SrdSurfsProperty.SkyViewFactor;
+            Surface.ViewFactorGroundIR = 1 - SrdSurfsViewFactor;
+            SrdSurfsProperty.GroundViewFactor = Surface.ViewFactorGroundIR;
+        } else if (SrdSurfsProperty.SkyViewFactor == 0.0 && SrdSurfsProperty.GroundViewFactor > 0.0) {
+            // If only ground view factor defined, sky view factor = 1 - all other defined view factors.
+            Surface.ViewFactorGroundIR = SrdSurfsProperty.GroundViewFactor;
+            Surface.ViewFactorSkyIR = 1 - SrdSurfsViewFactor;
+            SrdSurfsProperty.SkyViewFactor = Surface.ViewFactorSkyIR;
+        } else {
+            // If neither ground nor sky view factor specified, continue to use the original proportion.
```

Comment:
With the new code, if I am following correctly, the same inputs (`SkyViewFactor` is blank and `GroundViewFactor` is 0.0) will fall to this `else` block and the end result would be `ViewFactorGroundIR` and `ViewFactorSkyIR` proportioned.

Reply:
@mjwitte I agree, the proposed logic doesn't hold always. There are two options to fix logic: (1) add two flag variables that can be set to true if these view factor fields are blank, or (2) revert, the view factors initialization to -1. In both approaches we can keep the other code changes as is.

Reply:
@mjwitte Addressed using option 1 above.